src/components/questionStructure/Task/variants/HandwritingTask/live-updates.hook.ts

```diff
@@ -0,0 +1,160 @@
+/** Time to wait with user putting no stroks on the pages before API call */
+const NO_STROKES_WAIT_DELAY = 300
+
+/**
+ * The strokes object that is used to store the strokes from the Excalidraw canvas.
+ * Note that elements can be anything that Excalidraw supports. We filter as required.
+ */
+interface Strokes {
+  elements: SceneData['elements']
+}
+
+/**
+ * This represents what the hook exports to the outside world.
+ */
+interface LiveUpdateHook {
+  updateStrokes: (strokes: Strokes) => void
+}
+
+// MathPix API functions
+
+/**
+ * The token object that is used to authenticate with the Mathpix API.
+ */
+interface Token {
+  appToken: string
+  strokesSessionId: string
+}
+
+const transformStrokesForAPI = (strokes: Strokes): Record<string, number[][]> => {
+  const apiStrokes: Record<string, number[][]> = {
+    x: [],
+    y: [],
+  }
+
+  for (const element of strokes.elements ?? []) {
+    const newStrokeX: number[] = []
+    const newStrokeY: number[] = []
+
+    if (element.type !== 'freedraw') continue
+
+    for (const [x, y] of (element as ExcalidrawFreeDrawElement).points) {
+      newStrokeX.push(x + element.x)
+      newStrokeY.push(y + element.y)
+    }
+
+    apiStrokes.x.push(newStrokeX)
+    apiStrokes.y.push(newStrokeY)
+  }
+
+  return apiStrokes
+}
+
+const getLatexFromStrokes = (token: Token, strokes: Strokes, signal: AbortSignal) => {
+  return axios.post(
+    `https://api.mathpix.com/v3/strokes`,
+    {
+      strokes: {
+        strokes: transformStrokesForAPI(strokes),
+      },
+    },
+    {
+      signal: signal,
+      headers: {
+        app_token: token.appToken,
+        strokes_session_id: token.strokesSessionId,
+      },
+    }
+  )
+}
+
+const useLiveUpdates = (username: string, setLatex: (latex: string) => void): LiveUpdateHook => {
+  const [token, setToken] = useState<Token>({
+    appToken: '',
+    strokesSessionId: '',
+  })
+
+  const abortControllerRef = useRef<AbortController | null>(null)
+  const timeoutIdRef = useRef<ReturnType<typeof setTimeout> | null>(null)
+
+  // Get a token for a mathpix session for this user
+  const getToken = useCallback(() => {
+    axiosInstance.get(routes.getMathPixToken).then((res) => {
+      setToken(res.data)
+    })
+  }, [])
+
+  // Get token on startup (or username change)
+  useEffect(getToken, [getToken])
+
+  // useEffect to handle strokes changes
+  const updateStrokes = useCallback(
+    (strokes: Strokes) => {
+      // Cleanup previous actions
+      if (abortControllerRef.current) {
+        abortControllerRef.current.abort()
+      }
+      if (timeoutIdRef.current) {
+        clearTimeout(timeoutIdRef.current)
+      }
+
+      // Setup for new action
+      abortControllerRef.current = new AbortController()
+      const { signal } = abortControllerRef.current
+
+      // We use a timeout so that an API request once the person has stopped drawing for 300 ms
+      // this prevents constant requests to the API that can easier hit a rate limit
```

Comment:
This sentence is broken. Can it be rewritten?

Reply:
Rewritten in 69cf1de